presto-sqlserver/src/main/java/io/prestosql/plugin/sqlserver/SqlServerClient.java

```diff
@@ -428,8 +428,11 @@ public void setNull(PreparedStatement statement, int index)
         return handle.createQuery("" +
                 "SELECT data_compression_desc FROM sys.partitions p " +
                 "INNER JOIN sys.tables t ON p.object_id = t.object_id " +
+                "INNER JOIN sys.schemas s ON t.schema_id = s.schema_id " +
                 "INNER JOIN sys.indexes i ON t.object_id = i.object_id " +
-                "WHERE SCHEMA_NAME(t.schema_id) = :schema AND t.name = :table_name AND i.type IN (0,1) " +
+                "WHERE s.name = :schema AND t.name = :table_name " +
+                "AND p.index_id = 0 " + // Heap
```

Comment:
nit: this comment belongs to java now. Once sql code is copied to sql editor comment is lost. I still think that it has to be within sql.